src/utils/validators.js

```diff
@@ -0,0 +1,325 @@
+ *
+ * @throws {TypeError} Value should be a valid secret key.
+ */
+export function validateSecretKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidSecretKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'secret key',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a balance key.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid balance key.
+ */
+export function validateBalanceKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidBalanceKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'balance key',
+      fieldName
+    }))
+  }
+}
+
+/** Fields validators */
+
+/**
+ * Validate value to be a int64 amount.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {boolean} [allowZero] Is zero amount allowed.
+ * @param {number} [min] The minimum amount.
+ * @param {number} [max] The maximum amount.
+ *
+ * @throws {TypeError} Value should be a valid int64 amount.
+ */
+export function validateAmount ({
+  value,
+  fieldName = '',
+  allowZero = false,
+  max,
+  min
+}) {
+  if (!BaseOperation.isValidAmount(value, allowZero, max, min)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'should be a valid int64 amount',
+      params: { min, max },
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be an operation subject.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid operation subject.
+ */
+export function validateSubject ({ value, fieldName = '' }) {
+  if (!BaseOperation.isValidSubject(value)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'must be of type String 0-256 long',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be an asset code.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid asset code.
+ */
+export function validateAssetCode ({ value, fieldName = '' }) {
+  if (!BaseOperation.isValidAsset(value)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'must be valid asset code of type String 1-16 long',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a `xdr.FeeType` instance.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a `xdr.FeeType` instance.
+ */
+export function validateFeeType ({ value, fieldName = '' }) {
+  validateUndefined({ value, fieldName })
```

Comment:
undefined won't be instance of xdr.FeeType - it will fail on the next statement

Reply:
Removed